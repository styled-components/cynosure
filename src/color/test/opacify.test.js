// @flow
import opacify from '../opacify'

describe('opacify', () => {
  it('should increase the opacity of hex #fff by 0.1 and still be 1', () => {
    expect(opacify(0.1, '#fff')).toMatchSnapshot()
  })

  it('should increase the opacity of an 8-digit hex color by 0.1 and still be 1', () => {
    expect(opacify(0.1, '#6564CDB3')).toMatchSnapshot()
  })

  it('should increase the opacity of rgba(255, 0, 0, 0.5) by 0.1', () => {
    expect(opacify(0.1, 'rgba(101, 100, 205, 0.7)')).toMatchSnapshot()
  })

  it('should increase the opacity of rgba(255, 0, 0, .5) by 0.5', () => {
    expect(opacify(0.5, 'rgba(255, 0, 0, .5)')).toMatchSnapshot()
  })

  it('should increase the opacity of hsl(0, 0%, 100%) by 0.2 and still be 1', () => {
    expect(opacify(0.2, 'hsl(0, 0%, 100%)')).toMatchSnapshot()
  })

  it('should increase the opacity of hsla(0, 0%, 100%, .3) by 0.5', () => {
    expect(opacify(0.5, 'hsla(0, 0%, 100%, .3)')).toMatchSnapshot()
  })

  it('should not decrease the opacity below 0', () => {
    expect(opacify(-0.5, 'rgba(255, 0, 0, .2)')).toMatchSnapshot()
  })

  it('should not increase the opacity beyond 1', () => {
    expect(opacify(0.5, 'rgba(255, 0, 0, .8)')).toMatchSnapshot()
  })

  it('should increase the opacity when passed a string for amount', () => {
    expect(opacify('0.1', '#fff')).toMatchSnapshot()
  })

  it('should throw an error when enter an invalid color', () => {
    expect(() => {
      opacify(0.5, 'not a color')
    }).toThrow(new Error("Couldn't parse the color string. Please provide the color as a string in hex, rgb, rgba, hsl or hsla notation."))
  })
})
