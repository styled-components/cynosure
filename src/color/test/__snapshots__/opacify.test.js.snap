// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opacify should increase the opacity of an 8-digit hex color by 0.1 and still be 1 1`] = `"rgba(101,100,205,0.8)"`;

exports[`opacify should increase the opacity of hex #fff by 0.1 and still be 1 1`] = `"#fff"`;

exports[`opacify should increase the opacity of hsl(0, 0%, 100%) by 0.2 and still be 1 1`] = `"#fff"`;

exports[`opacify should increase the opacity of hsla(0, 0%, 100%, .3) by 0.5 1`] = `"rgba(255,255,255,0.8)"`;

exports[`opacify should increase the opacity of rgba(255, 0, 0, .5) by 0.5 1`] = `"#f00"`;

exports[`opacify should increase the opacity of rgba(255, 0, 0, 0.5) by 0.1 1`] = `"rgba(101,100,205,0.8)"`;

exports[`opacify should increase the opacity when passed a string for amount 1`] = `"#fff"`;

exports[`opacify should not decrease the opacity below 0 1`] = `"rgba(255,0,0,0)"`;

exports[`opacify should not increase the opacity beyond 1 1`] = `"#f00"`;
